retriever/lib/engine_tools.py

```diff
@@ -75,6 +75,16 @@ def name_matches(scripts, arg):
         if arg == script.name.lower():
             return [script]
 
+    from retriever.lib.scripts import get_script_upstream
+
+    if arg in RETRIEVER_DATASETS:
+        read_script = get_script_upstream(arg, repo=RETRIEVER_REPOSITORY)
+    else:
+        read_script = get_script_upstream(arg)
+
+    if read_script is not None:
+        return [read_script]
+
```

Comment:
First checks if the script is available locally, if present returns it. Then, checks if it is a retriever legacy script or not. Then downloads it from the according upstream repository.